allocation_manager/service/unit.go

```diff
@@ -46,18 +49,56 @@ metricLicenseReleaseReason = promauto.NewCounterVec(prometheus.CounterOpts{
 	},
 )
 */
+
+unitCounter = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "unit_operations_total",
+		Help: "Total number of operations performed on units",
+	},
+	[]string{"kind", "unit", "operation"},
+)
 )
 
+func init() {
+	fmt.Println("In init")
+	prometheus.MustRegister(unitCounter)
+}
+
+func (u unit) DoOperation(operationName string) {
+	// name := u.name
+	name := u.Topology.Name
+	unitCounter.With(prometheus.Labels{
+		"kind": fmt.Sprintf("%T", u),
```

Comment:
The risk with using %T (i.e. "service.unit") is in case of refactoring, the metric labels will change which is not desirable. Also, it arguably duplicated in the counter name "unit_..."